### ISSUE TITLE
output_types with a space between the types crashes multi_scan

### ISSUE THREAD

**Issue by selin194** [lyndsey-ferguson/fastlane-plugin-test_center#182]
### Question Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I searched for [existing GitHub issues](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues)

### Question Subject
How to generate report.html, report.xml, report.junit and report.xcresult files at once?

#### Question Description

I am trying to execute ios UI tests with 2 simulators. And I want to generate report.html, report.junit, report.xml and report.xcresult files in Xcode 11.  When I use only output file option like below: 

`            result = multi_scan(
                      workspace: 'udemy.xcworkspace',
                      try_count: 3,
                      fail_build: false,
                      scheme: 'AcceptanceTests',
                      device: 'iPhone 8 Plus',
                      force_quit_simulator: true,
                      reset_simulator: true,
                      fail_build: false,
                      testrun_completed_block: test_run_block,
                      parallel_testrun_count: 2,
                      output_types: 'junit',
                      output_files: 'report.xml'
               )      
`
`report.xml` file is successfully generated. However I want to produce other artifact files that are : report.junit, report.xml, report.html and report.xcresult files.
I already tried to give these arguments like below:

`            result = multi_scan(
                      workspace: 'udemy.xcworkspace',
                      try_count: 3,
                      fail_build: false,
                      scheme: 'AcceptanceTests',
                      device: 'iPhone 8 Plus',
                      force_quit_simulator: true,
                      reset_simulator: true,
                      fail_build: false,
                      testrun_completed_block: test_run_block,
                      parallel_testrun_count: 2,
                      output_types: 'junit, html',
                      output_files: 'report.xml, report.html'
               )      
`
However with this simple configuration I received error like this and I did not know how to generate all the files at once by running tests parallel.
The error was:
`ERROR [2019-12-17 08:32:24.99]: [0;31;49mno implicit conversion from nil to integer[0m`
<img width="1236" alt="Screen Shot 2019-12-17 at 20 49 28" src="https://user-images.githubusercontent.com/39696200/71021159-cf560000-210e-11ea-9dc6-1e8b1bf3f5cd.png">

So my question is : How can I generate all of the files (junit,html,xml and xcresult) as artifacts while running tests in parallel with 2 simulators? Thanks in advance! @lyndsey-ferguson 

NOTE: Xcode version : `11.2.1`, test_center version: `3.8.15` , jenkins version: `2.176.3`

**Comment by lyndsey-ferguson** [comment]
Hello @selin194, you cannot provide `xcresult` files in the `output_types` yet. That is part of #164 that is in slow progress. You can provide `result_bundle: true` as described in the underlying action's [options](https://docs.fastlane.tools/actions/scan/#parameters) (scan).

However, those xcresults will **not** be collated. If this answers your question, I'll close this Issue and suggest you keep an eye on the other one. The PR is #167

**Comment by lyndsey-ferguson** [comment]
Oh, and the `xml` file type is the same as `junit`.

**Comment by selin194** [comment]
I see, thank you for responding so quickly. However I can not give `html` and `junit` output types at the same time. By giving:
`output_types: 'junit, html', output_files: 'report.xml, report.html'` which gives me error when I run the jobs.

**Comment by lyndsey-ferguson** [comment]
Ah, try without the space. That shouldn't cause an issue, but that is what jumps out at me.
`output_types: 'junit,html', output_files: 'report.xml,report.html'`. If that fixes it, we can create a bug Issue for this as the spaces should not be a problem.

But you can see from [here](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/spec/reportname_helper_spec.rb#L93) that without spaces in my unit tests works.

**Comment by selin194** [comment]
Hey again, I tried `without spaces` and  I can successfully take both `xml` and `html` reports.

**Comment by lyndsey-ferguson** [comment]
Thanks @selin194, I've updated this Issue to describe the problem. For now, please remove the extra space.

**Comment by selin194** [comment]
Thank you @lyndsey-ferguson 🎉

**Comment by lyndsey-ferguson** [comment]
@selin194, I believe that I have fixed the issue. Can you add the spaces back to your `:output_types` parameter, modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-182-fix-output-types-with-spaces"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by selin194** [comment]
Hey @lyndsey-ferguson, I followed the steps that you wrote and it worked with spaces!
When I give output types and files as below with specified branch `gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-182-fix-output-types-with-spaces"` 
```  
output_types: 'junit, html',
output_files: 'report.xml, report.html'
```
Output files are successfully generated. Thanks for this fix 🎉

**Comment by lyndsey-ferguson** [comment]
You're welcome, thank you for confirming the fix.

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.8.16](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.8.16)
